tests/OSPSuite.Core.Tests/Domain/FormulaTaskSpecs.cs

```diff
@@ -489,17 +489,206 @@ protected override void Context()
          //..|<NBH>|..|..|Plasma|<NBH>|K_<==_does_not_exist_in_the_model
-         _objectPath = new FormulaUsablePath(PARENT_CONTAINER, NBH, PARENT_CONTAINER, PARENT_CONTAINER, "Plasma", NBH, _neighborhoodParameter.Name) { Alias = "K" };
+         _objectPath = new FormulaUsablePath(PARENT_CONTAINER, NBH, PARENT_CONTAINER, PARENT_CONTAINER, "Plasma", NBH, _neighborhoodParameter.Name) {Alias = "K"};
          _liverCellParameter.Formula.AddObjectPath(_objectPath);
       }
 
       [Observation]
-      public void should_have_replaced_the_nbh_with_the_actual_path_to_the_neighborhood()
+      public void should_throw_an_exception()
       {
          The.Action(() => sut.ExpandNeighborhoodReferencesIn(_model)).ShouldThrowAn<OSPSuiteException>();
       }
    }
 
+   public abstract class concern_for_lumen_segment_path_resolution : concern_for_FormulaTask
+   {
+      protected FormulaUsablePath _objectPath;
+      protected Container _smallIntestine;
+      protected Model _model;
+      protected Container _mucosa;
+      protected Container _duodenumMucosa;
+      protected Container _organism;
+      protected Container _lumen;
+      protected Container _duodenumLumen;
+      protected IParameter _volumeDuodenumLumen;
+      protected Container _duodenumMucosaIntracellular;
+
+      protected override void Context()
+      {
+         base.Context();
+         _model = new Model();
+         _organism = new Container().WithName(ORGANISM);
+         _smallIntestine = new Container().WithName("SmallIntestine").Under(_organism);
+         _mucosa = new Container().WithName("Mucosa").Under(_smallIntestine);
+         _duodenumMucosa = new Container().WithName("Duodenum").Under(_mucosa);
+         _duodenumMucosaIntracellular = new Container().WithName("Intracellular").Under(_duodenumMucosa);
+
+         _lumen = new Container().WithName(LUMEN).Under(_organism);
+         _duodenumLumen = new Container().WithName("Duodenum").Under(_lumen);
+         _volumeDuodenumLumen = DomainHelperForSpecs.ConstantParameterWithValue(10).WithName(VOLUME).Under(_duodenumLumen);
+
+         _model.Root = _organism;
+      }
+   }
+
+   public class When_replacing_the_lumen_segment_keyword_in_a_well_defined_absolute_path : concern_for_lumen_segment_path_resolution
+   {
+      protected Parameter _parameterReferencingAbsoluteLumenSegment;
+
+      protected override void Context()
+      {
+         base.Context();
+
+         _parameterReferencingAbsoluteLumenSegment = new Parameter().WithName("P").WithParentContainer(_organism);
+         _parameterReferencingAbsoluteLumenSegment.Formula = new ExplicitFormula("V+10");
+         //Organism|SmallIntestine|Mucosa|Duodenum|LumenSegment|V
+         _objectPath = new FormulaUsablePath(ORGANISM, _smallIntestine.Name, _mucosa.Name, _duodenumMucosa.Name, LUMEN_SEGMENT, _volumeDuodenumLumen.Name) {Alias = "V"};
+         _parameterReferencingAbsoluteLumenSegment.Formula.AddObjectPath(_objectPath);
```

Comment:
let's add a second object path to the formula, which does not contain LUMEN_SEGMENT 😉

Reply:
done